### PR TITLE
Test Comp

### PR DESCRIPTION
**Pull request by flore77**
## Need

```js
As a developer
I need a test comp
So that I can present the result of the test
```

## Deliverables
A test component that presents the state, i.e pass or fail, and other details, i.e baseline/screenshot/diff or Error etc.

## Solution

The `Test comp` will display the test title and its state. 

The other details will be displayed by another component, that will be rendered by the Test comp through a transition, like [jQuery toggle](http://api.jquery.com/toggle/), but using `CSS3 transition`.

### CODE REVIEW

**Comment by flore77** [comment]
@andrei-picus-hs please review my code.

**Comment by flore77** [comment]
@andrei-picus-hs 
dm

File: tests/unit/components/test.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  it('should display the test title', function() {
+    expect(tree.textIn('.test-title')).to.contain(passTest.title);
+  });
+
+  it('should display the pass state', function() {
+    expect(tree.textIn('.test-state')).to.be.equal(passTest.state);
+  });
+
+  it('should display the fail state', function() {
+    tree = sd.shallowRender(<Test test={failTest}/>);
```
Comment (andrei-picus-hs) [inline tests/unit/components/test.js]: Split this into a separate test file, or at least into a separate describe with another `beforeEach` block.

File: tests/fixtures/components/test.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+module.exports = [{
+  title: 'Test 1',
+  state: 'passed',
+  result: {
+    isEqual: true,
+    baseline: 'baselinePath 1'
+  },
+  duration: 7
+}, {
```
Comment (andrei-picus-hs) [inline tests/fixtures/components/test.js]: Create 2 separate fixtures for this.

File: tests/unit/components/test.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+var sd = require('skin-deep'),
+    expect = require('chai').expect,
+    Test = require('../../../ui/components/test.jsx'),
+    fixture = require('../../fixtures/components/test.js');
+
+describe('Test', function() {
+  var tree, passTest, failTest;
+
+  before(function() {
+    passTest = fixture[0];
```
Comment (andrei-picus-hs) [inline tests/unit/components/test.js]: See [above](https://github.com/uberVU/mocha-mugshot-reporter/pull/14/files#r40783931).

File: ui/components/test.jsx [hunk 4]
```diff
@@ -1,8 +1,30 @@
...
+        classes = cx({
+          test: true,
+          toggled: this.state.toggled
+        });
+
+    return <div className={classes}>
+      <p className='test-title' onClick={this.displayDetails}>{test.title}
+        <span className='test-state'>{test.state}</span>
+      </p>
+      <Details/>
```
Comment (andrei-picus-hs) [inline ui/components/test.jsx]: Shouldn't you be sending some props to this component? I didn't see any test for this.

**Comment by andrei-picus-hs** [comment]
Finished review.

File: ui/components/test.jsx (hunk 4 above)
Comment (flore77) [inline ui/components/test.jsx]: Next issue, #17 !!

Comment (andrei-picus-hs) [inline ui/components/test.jsx]: A good practice is to start using the layer below you before you implement it. It makes you sit and define your interface for that layer. In React, you can safely send props to a component that won't use them.

You can leave it like this and I'll review it in that other issue :smile:

Comment (flore77) [inline ui/components/test.jsx]: Ok. Got it!!

**Comment by flore77** [comment]
@andrei-picus-hs please review

**Comment by andrei-picus-hs** [comment]
LGTM

**Pull request merged**